src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,152 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        String query = "INSERT INTO shopping_carts (user_id) VALUES (?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement(query, PreparedStatement.RETURN_GENERATED_KEYS);
+            preparedStatement.setLong(1, shoppingCart.getUserId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                shoppingCart.setId(resultSet.getLong(1));
+            }
+            return shoppingCart;
+        } catch (SQLException e) {
+            throw new RuntimeException("Unable to create shopping cart: ", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts WHERE cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                ShoppingCart shoppingCart = getShoppingCartFromResultSet(resultSet);
```

Comment:
```suggestion
                var shoppingCart = getShoppingCartFromResultSet(resultSet);
```
we can use var here, because we are declaring variable, and only on the next line we are using it